astoria/common/disks/type_calculator.py

```diff
@@ -0,0 +1,36 @@
+"""Class to determine the type of a disk."""
+
+from pathlib import Path
+from typing import Dict
+
+from .constraints import Constraint, FilePresentConstraint, TrueConstraint
+from .structs import DiskType
+
+
+class DiskTypeCalculator:
+    """
+    Helper class to calculate the DiskType of a given disk drive.
+
+    .. warning:: A disk must only rely on itself and not other disks to determine it
+             the type of the disk.
+    """
+
+    def calculate(self, path: Path) -> DiskType:
+        """
+        Calculate the DiskType of a drive given it's mount path.
+
+        :param path: The mount path of the drive.
+        :returns: The type of the disk.
+        """
+        constraints: Dict['DiskType', Constraint] = {
+            DiskType.USERCODE: FilePresentConstraint("robot.zip"),
+            DiskType.METADATA: FilePresentConstraint("astoria.json"),
```

Comment:
Thought: could also add a constraint on it being valid JSON

Reply:
Spun out into #152